applications/DEMApplication/custom_constitutive/DEM_rolling_friction_model_constant_torque.cpp

```diff
@@ -29,56 +32,50 @@ namespace Kratos{
             GeometryFunctions::normalize(elementRelAngularVelocity);
-
             array_1d<double, 3> elementRelAngularVelocity_normalise = elementRelAngularVelocity;
 
-            Properties& properties_of_this_contact = p_element->GetProperties().GetSubProperties(p_neighbor->GetProperties().Id());
-
-            mContactMoment[0] -= elementRelAngularVelocity_normalise[0] * fabs(LocalContactForce[2]) * contact_center_point_to_element1_mass_center_distance * properties_of_this_contact[ROLLING_FRICTION]; 
-
-            mContactMoment[1] -= elementRelAngularVelocity_normalise[1] * fabs(LocalContactForce[2]) * contact_center_point_to_element1_mass_center_distance * properties_of_this_contact[ROLLING_FRICTION]; 
+            // Get equivalent radius (this only works for sphere particles)
+            const double my_radius    = p_element->GetRadius();
+            const double other_radius = p_neighbor->GetRadius();
+            const double equiv_radius = my_radius * other_radius / (my_radius + other_radius);
 
-            mContactMoment[2] -= elementRelAngularVelocity_normalise[2] * fabs(LocalContactForce[2]) * contact_center_point_to_element1_mass_center_distance * properties_of_this_contact[ROLLING_FRICTION]; 
+            // Get rolling friction coefficient
+            Properties& r_properties = p_element->GetProperties().GetSubProperties(p_neighbor->GetProperties().Id());
+            const double rolling_friction_coefficient = r_properties[ROLLING_FRICTION];
 
-        } 
+            mContactMoment[0] -= elementRelAngularVelocity_normalise[0] * std::abs(LocalContactForce[2]) * equiv_radius * rolling_friction_coefficient;
+            mContactMoment[1] -= elementRelAngularVelocity_normalise[1] * std::abs(LocalContactForce[2]) * equiv_radius * rolling_friction_coefficient;
+            mContactMoment[2] -= elementRelAngularVelocity_normalise[2] * std::abs(LocalContactForce[2]) * equiv_radius * rolling_friction_coefficient;
+        }
     }
 
-    void DEMRollingFrictionModelConstantTorque::ComputeRollingFrictionWithWall(double LocalContactForce[3], SphericParticle* p_element, Condition* const wall, double indentation, array_1d<double, 3>& mContactMoment)
+    void DEMRollingFrictionModelConstantTorque::ComputeRollingFrictionWithWall(SphericParticle* p_element, Condition* const wall, double LocalCoordSystem_2[3], double LocalContactForce[3], array_1d<double, 3>& mContactMoment)
     {
         array_1d<double, 3> element1AngularVelocity;
         noalias(element1AngularVelocity) = p_element->GetGeometry()[0].FastGetSolutionStepValue(ANGULAR_VELOCITY);
         if (element1AngularVelocity[0] || element1AngularVelocity[1] || element1AngularVelocity[2]){
-
-            const double my_young      = p_element->GetYoung(); 
-            const double walls_young   = wall->GetProperties()[YOUNG_MODULUS];
-            
-            double arm_length = p_element->GetInteractionRadius() - indentation* walls_young / (walls_young + my_young); 
```

Comment:
For particle-wall interaction, maybe the old version of arm_length which takes the indentation into account is more reasonable. Sometimes, literature does not mean perfect. We can discuss this point more @rlrangel.

Reply:
I agree that discounting the indentation from the radius is more consistent with the geometry of the simulation, although in the literature, in general, I don't see this being taken into account. We can discount the indentation from the particle-wall contact, but then, to be consistent, we also need to do the same in the particle-particle contact.

Reply:
I agree.